src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioTaskSchedulerFactory.cs

```diff
@@ -72,5 +76,38 @@ public Task<T> ScheduleTask<T>(Func<Task<T>> taskFunc, string taskName, Cancella
                 return _queue.Value.ScheduleTask(taskFunc, taskName, cancellationToken);
             }
         }
+
+        private class JoinableTaskFactoryTaskScheduler : TaskScheduler
+        {
+            private readonly JoinableTaskFactory _joinableTaskFactory;
+
+            public JoinableTaskFactoryTaskScheduler(JoinableTaskFactory joinableTaskFactory)
+            {
+                _joinableTaskFactory = joinableTaskFactory;
+            }
+
+            public override int MaximumConcurrencyLevel => 1;
+
+            protected override IEnumerable<Task> GetScheduledTasks() => null;
+
+            protected override void QueueTask(Task task)
+            {
+                _joinableTaskFactory.RunAsync(async () =>
```

Comment:
This bit seems somewhat unfortunate: we get to kick a task off to the thread pool just to jump back...for every task.

Reply:
➡️ Very much unlike `Task.Run`, `JoinableTaskFactory.RunAsync` does not kick of a task. It merely acts as the entry point to an asynchronous operation when called from a synchronous method.